src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -99,10 +100,11 @@ citus_table_is_visible(PG_FUNCTION_ARGS)
 
 /*
  * ErrorIfRelationIsAKnownShard errors out if the relation with relationId is
- * a shard relation.
+ * a shard relation. Provides an additional info about the GUC
+ * allow_manual_changes_to_shards if bool manualChangeDisabled is true.
  */
 void
-ErrorIfRelationIsAKnownShard(Oid relationId)
+ErrorIfRelationIsAKnownShard(Oid relationId, bool manualChangeDisabled)
```

Comment:
the meaning of this argument is not very obvious to me. It might be nicer to introduce a separate function, e.g. `ErrorIfChangingKnownShard(relationId);`

We can then check the GUC in the function itself, even before checking whether it's a known shard.